File: Atributos de validación Data Annotations multiidioma/[C#]-Atributos de validación Data Annotations multiidioma/C#/LocalizeDataAnnotations/Infrastructure/CustomRequiredAttribute.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace LocalizeDataAnnotations.Infrastructure
{
    public class CustomRequiredAttribute: RequiredAttribute
    {

        public override string FormatErrorMessage(string name)
        {
            string errorMessage = null;
            if (!string.IsNullOrEmpty(base.ErrorMessageResourceName))
                errorMessage = TextosBBDD.Recuperar(base.ErrorMessageResourceName);

            return errorMessage ?? base.FormatErrorMessage(name);
        }

    }
}

File: Atributos de validación Data Annotations multiidioma/[VB]-Atributos de validación Data Annotations multiidioma/VB/LocalizeDataAnnotations/Infrastructure/CustomRequiredAttribute.vb
Imports System.ComponentModel.DataAnnotations

Public Class CustomRequiredAttribute
    Inherits RequiredAttribute

    Public Overrides Function FormatErrorMessage(name As String) As String
        Dim errorMessage As String = Nothing
        If (Not String.IsNullOrEmpty(MyBase.ErrorMessageResourceName)) Then
            errorMessage = TextosBBDD.Recuperar(MyBase.ErrorMessageResourceName)
        End If
        Return If(errorMessage, MyBase.FormatErrorMessage(name))
    End Function

End Class
